jaxrs-api/src/main/java/jakarta/ws/rs/ext/EntityPart.java

```diff
@@ -0,0 +1,274 @@
+         * also set the {@code Content-Type} header for this part. This call is
+         * effectively the same as
+         * {@code mediaType(MediaType.valueOf(mediaTypeString))}.
+         * 
+         * @param mediaTypeString the media type for the part to be built
+         * @return the updated builder
+         * @throws IllegalArgumentException if {@code mediaTypeString} cannot be parsed
+         *                                  or is {@code null}
+         */
+        public Builder mediaType(String mediaTypeString) throws IllegalArgumentException;
+
+        /**
+         * Adds a new header or replaces a previously added header and sets the header value(s).
+         * 
+         * @param headerName the header name
+         * @param headerValues the header value(s)
+         * @return the updated builder
+         * @throws IllegalArgumentException if {@code headerName} is {@code null}
+         */
+        public Builder header(String headerName, String... headerValues) throws IllegalArgumentException;
+
+        /**
+         * Adds new headers or replaces previously added headers. The behavior of this method would be the same as if
+         * iterating over the entry set and invoking the {@link #header(String, String...)} method.
+         * 
+         * @param newHeaders the multivalued map of headers to add to this part
+         * @return the updated builder
+         * @throws IllegalArgumentException if {@code newHeaders} is {@code null}
+         */
+        public Builder headers(MultivaluedMap<String, String> newHeaders) throws IllegalArgumentException;
+
+        /**
+         * Sets the file name for this part. The file name will be specified as an attribute in the
+         * {@code Content-Disposition} header of this part. When this method is called, the default
+         * media type used for the built part will be "application/octet-stream" if not otherwise
+         * specified.
+         * 
+         * @param fileName the file name for this part
+         * @return the updated builder
+         * @throws IllegalArgumentException if {@code fileName} is {@code null}
+         */
+        public Builder fileName(String fileName) throws IllegalArgumentException;
+
+        /**
+         * Sets the content for this part. This method or the convenience method,
+         * {@link #content(String, InputStream)} must be invoked before invoking the
+         * {@link #build()} method.
+         * 
+         * The {@code InputStream} will be closed by the implementation code after
+         * sending the multipart data. Closing the stream before it is sent could
+         * result in unexpected behavior.
+         * 
+         * @param content {@code InputStream} of the content of this part
+         * @return the updated builder
+         * @throws IllegalArgumentException if {@code content} is {@code null}
+         */
+        public Builder content(InputStream content) throws IllegalArgumentException;
```

Comment:
I find allowing only `InputStream` based content too limiting. Content of a part can quite often be just a String -- a form field in usual multipart HTML form, or the user might want to map an in-memory object to a JSON or other format and send that. Allowing just `InputStream` would make these usacases too cumbersome:

```
String userName;
EntityPart userNamePart = EntityPart.withName("userName")
    .mediaType("text/plain;charset=UTF-8")
    .content(new ByteArrayInputStream(
        userName.getBytes(StandardCharsets.UTF_8))
    .build();
```

The API here should allow full usage of `MessageBodyWriter` capabilities of JAX-RS with InputStream being just a special case:

```
String userName;
EntityPart userNamePart = EntityPart.withName("userName")
    .mediaType("text/plain;charset=UTF-8")
    .content(userName)
   .build();
```

```
JsonObject metrics;
EntityPart metricsPart = EntityPart.withFileName("metrics.json")
    .mediaType(MediaType.APPLICATION_JSON)
    .content(metrics)
    .build();
````

Reply:
Thanks for the suggestion!  I added these two new methods:
```
public <T> Builder content(T content, Class<? extends T> type) throws IllegalArgumentException;
public <T> Builder content(T content, GenericType<T> type) throws IllegalArgumentException;
```

The type parameter should be required in order to find an applicable MBW.  For example, you might have a MBW that implements `MessageBodyWriter<MyInterface>`, but pass in an instance of `MyImpl` - or perhaps a dynamic proxy of `MyInterface`.

Reply:
It wouldn't hurt to serve the most usual cause of `content(entity, entity.getClass())` with a (default) method.

On the other hand, if we look at `Response.ResponseBuilder`, it offers following methods for respone content (entity):

```
        public abstract ResponseBuilder entity(Object entity);
        public abstract ResponseBuilder entity(Object entity, Annotation[] annotations);
```

Where handling of generic types is done by utilizing `GenericEntity<T>`. I believe similar content methods would provide least surprise as EntityPart would work exactly like response entity. These methods also add option for adding  explicit annotations for `MessageBodyWriter`, which would require even more overloads in current proposal to cover.

Reply:
Sounds good.  I added a default method `content(Object)` that delegates to `content(o, o.getClass())`.  Thanks for the suggestion!